scripts/download-wda-sim.mjs

```diff
@@ -0,0 +1,76 @@
+import {fs, logger, zip, net, node} from 'appium/support.js';
+import _ from 'lodash';
+import os from 'os';
+import path from 'path';
+import { parseArgValue } from './utils.mjs';
+
+const log = logger.getLogger('WDA');
+const wdaUrl = (version, zipFileName) =>
+  `https://github.com/appium/WebDriverAgent/releases/download/v${version}/${zipFileName}`;
+const destZip = (platform) => {
+  const scheme = `WebDriverAgentRunner${_.toLower(platform) === 'tvos' ? '_tvOS' : ''}`;
+  return `${scheme}-Build-Sim-${os.arch() === 'arm64' ? 'arm64' : 'x86_64'}.zip`;
+};
+
+/**
+ * Return installed appium-webdriveragent package version
+ * @returns {number}
+ */
+async function webdriveragentPkgVersion() {
+  const pkgPath = path.join(
+    node.getModuleRootSync('appium-xcuitest-driver', import.meta.url),
+    'node_modules',
+    'appium-webdriveragent',
+    'package.json'
+  );
+  return JSON.parse(await fs.readFile(pkgPath, 'utf8')).version;
+};
+
+/**
+ * Prepare the workng root directory.
+ * @returns {string} Root directory to download and unzip.
+ */
+async function prepareRootDir() {
+  const destDirRoot = parseArgValue('outdir');
+  if (!destDirRoot) {
+    log.error(`--outdir is required`);
+    throw new Error();
```

Comment:
Either add the error description to the error message or exit with non-zero exit code after the corresponding error log